adapters/timm/timm_model_adapter.py

```diff
@@ -0,0 +1,358 @@
+import os
+import timm
+import torch
+from tqdm import tqdm
+import torchvision.transforms as transforms
+import dtlpy as dl
+import time
+from torch.utils.data import DataLoader
+from imgaug import augmenters as iaa
+import torch.optim as optim
+import torch.nn.functional
+import numpy as np
+import torch.nn
+import logging
+import copy
+from dtlpy.utilities.dataset_generators.dataset_generator import collate_torch
+from dtlpy.utilities.dataset_generators.dataset_generator_torch import DatasetGeneratorTorch
+
+logger = logging.getLogger('MobileNetV3-adapter')
+
+
+@dl.Package.decorators.module(name='model-adapter',
+                              description='Model Adapter for TIMM Models',
+                              init_inputs={'model_entity': dl.Model})
+class TIMMAdapter(dl.BaseModelAdapter):
+    """
+    MobileNetV3 Model adapter using Pytorch.
+    The class bind Dataloop model and model entities with model code implementation
+    """
+    def __init__(self, model_entity=None):
+        self.device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+        self.model_name = self.configuration.get('model_name')
+        super(TIMMAdapter, self).__init__(model_entity=model_entity)
+
+    def load(self, local_path, **kwargs):
+        """ Loads model and populates self.model with a `runnable` model
+
+            This function is called by load_from_model (download to local and then loads)
+
+        :param local_path: `str` directory path in local FileSystem
+        """
+        weights_filename = self.model_entity.configuration.get('weights_filename')
+        model_path = str(os.path.join(local_path, weights_filename))
+        if weights_filename and os.path.exists(model_path):
+            # With the model entity and weights from the platform
+
+            logger.info(f"Loading a model from {model_path}")
+            self.model = torch.load(model_path, map_location=self.device)
+            self.model.to(self.device)
+            self.model.eval()
+            logger.info(f"Loaded model from {model_path} successfully")
+        else:
+            # With the model from the library
+            self.model = timm.create_model(self.model_name, pretrained=True)
+            self.model.to(self.device)
+            self.model.eval()
+            logger.info(f"Loaded model from library successfully")
```

Comment:
self.model.to(self.device)
            self.model.eval()
            
       can be outside if and else